sdk/eventhub/azure-eventhub/azure/eventhub/_common.py

```diff
@@ -165,11 +178,24 @@ def _from_message(cls, message):
         """
         event_data = cls(body="")
         event_data.message = message
+        event_data._raw_amqp_message = AmqpAnnotatedMessage(message=message)
```

Comment:
My concern is that it waste cycles for constructing the ivar raw_amqp_message and message of EventData twice in the sending scenario.

I think we could update this method to take an optional raw_amqp_message kwargs so that in the sending case, we pass the both the  AMQPAnnoatedMessage and the AMQPAnnoatedMessage._to_outgoing_amqp_message

e.g. code in service bus.

https://github.com/Azure/azure-sdk-for-python/blob/3a471ada95c696fd2c3795f80a46cc381f977067/sdk/servicebus/azure-servicebus/azure/servicebus/amqp/_amqp_message.py#L296-L298

https://github.com/Azure/azure-sdk-for-python/blob/3a471ada95c696fd2c3795f80a46cc381f977067/sdk/servicebus/azure-servicebus/azure/servicebus/_common/message.py#L101-L104

Reply:
If we do the above as you mentioned, then we would need to add kwargs to `EventData` constructor. Is this what we want to do @annatisch ? (I believe this also ties into the comment you made above:  https://github.com/Azure/azure-sdk-for-python/pull/19287#discussion_r656515493)

Reply:
for now, just adding extra param to `EventData._from_message()`. Will think about `kwargs` later